[PyTorch][HAM]: Deprecate functionalize

XuanQi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103053

TLDR is to remove this flag and utilize functionalization functionality by lower level lib directly.

Differential Revision: [D46469134](https://our.internmc.facebook.com/intern/diff/D46469134/)

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy

XuanQi committed:
[PyTorch][HAM]: Deprecate functionalize

TLDR is to remove this flag and utilize functionalization functionality by lower level lib directly.

Differential Revision: [D46469134](https://our.internmc.facebook.com/intern/diff/D46469134/)

[ghstack-poisoned]

tugsbayasgalan commented on torch/_dynamo/eval_frame.py:
```diff
@@ -1015,19 +1005,7 @@ def from_fun(t):
-                if functionalize:
-                    torch._enable_functionalization(reapply_views=True)
-                    try:
-                        return pytree.tree_map(
```
Can you delete to_fun and from_fun as well?

tugsbayasgalan reviewed (approved):
Looks good! Just small comment

XuanQi committed:
Update on "[PyTorch][HAM]: Deprecate functionalize"


TLDR is to remove this flag and utilize functionalization functionality by lower level lib directly.

Differential Revision: [D46469134](https://our.internmc.facebook.com/intern/diff/D46469134/)

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy

[ghstack-poisoned]

XuanQi committed:
Update on "[PyTorch][HAM]: Deprecate functionalize"


TLDR is to remove this flag and utilize functionalization functionality by lower level lib directly.

Differential Revision: [D46469134](https://our.internmc.facebook.com/intern/diff/D46469134/)

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy

[ghstack-poisoned]

XuanQi committed:
Update on "[PyTorch][HAM]: Deprecate functionalize"


TLDR is to remove this flag and utilize functionalization functionality by lower level lib directly.

Differential Revision: [D46469134](https://our.internmc.facebook.com/intern/diff/D46469134/)

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy

[ghstack-poisoned]
